dev/tests/integration/testsuite/Magento/MediaGallery/Model/ResourceModel/AssetKeywordsTest.php

```diff
@@ -66,45 +66,61 @@ protected function setUp(): void
      *
      * @magentoDataFixture Magento/MediaGallery/_files/media_asset.php
      * @dataProvider keywordsProvider
-     * @param array $keywords
+     * @param string[] $keywords
+     * @param string[] $updatedKeywords
      * @throws \Magento\Framework\Exception\LocalizedException
      */
-    public function testSaveAndGetKeywords(array $keywords): void
+    public function testSaveAndGetKeywords(array $keywords = [], array $updatedKeywords = []): void
     {
-        $keywords = ['pear', 'plum'];
-
         $loadedAssets = $this->getAssetsByPath->execute([self::FIXTURE_ASSET_PATH]);
         $this->assertCount(1, $loadedAssets);
         $loadedAsset = current($loadedAssets);
 
+        $this->updateAssetKeywords($loadedAsset->getId(), $keywords);
+        $this->updateAssetKeywords($loadedAsset->getId(), $updatedKeywords, $keywords);
+    }
+
+    /**
+     * Update Asset keywords
+     *
+     * @param int $assetId
+     * @param string[] $keywords
+     * @param string[] $currentKeywords
+     */
+    private function updateAssetKeywords(int $assetId, array $keywords = [], array $currentKeywords = []): void
```

Comment:
I don't think we need the `$currentKeywords` parameter for this function at all, only the actual (new) keywords should be used for update and verification